src/util/commandUtils.ts

```diff
@@ -0,0 +1,255 @@
+        return formDefaultCommand(gradleCmdStart, buildGradlePath, command, "-b=", customCommand);
+    }
+}
+
+/**
+ * Reused from vscode-maven
+ * https://github.com/microsoft/vscode-maven/blob/main/src/mavenTerminal.ts
+ */
+function toDefaultWslPath(p: string): string {
+    const arr: string[] = p.split(":\\");
+    if (arr.length === 2) {
+        const drive: string = arr[0].toLowerCase();
+        const dir: string = arr[1].replace(/\\/g, "/");
+        return `/mnt/${drive}/${dir}`;
+    } else {
+        return p.replace(/\\/g, "/");
+    }
+}
+
+/**
+ * Reused from vscode-maven
+ * https://github.com/microsoft/vscode-maven/blob/2ab8f392f418c8e0fe2903387f2b0013a1c50e78/src/utils/mavenUtils.ts
+ */
+export function isWin(): boolean {
+    return process.platform.startsWith("win");
+}
+
+/**
+ * Returns maven wrapper path or mvn
+ */
+async function mvnCmd(pomPath: string): Promise<string> {
+    const preferMavenWrapper: boolean | undefined = vscode.workspace.getConfiguration("maven").get<boolean>("executable.preferMavenWrapper");
+    if (preferMavenWrapper) {
+        const localMvnwPath: string | undefined = await getLocalMavenWrapper(Path.dirname(pomPath));
+        if (localMvnwPath) {
+            return `${localMvnwPath}`;
+        }
+    }
+    return "mvn";
+}
+
+/**
+ * Returns gradle wrapper path or gradle
+ */
+async function gradleCmd(buildGradle: string): Promise<string> {
+    const preferGradleWrapper: boolean | undefined = vscode.workspace.getConfiguration("java").get<boolean>("import.gradle.wrapper.enabled");
+    if (preferGradleWrapper) {
+        const localGradlewPath: string | undefined = await getLocalGradleWrapper(Path.dirname(buildGradle));
+        if (localGradlewPath) {
+            return `${localGradlewPath}`;
+        }
+    }
+    return "gradle";
+}
+
+/**
+ * Search for potential Maven wrapper, return undefined if does not exist
+ * Reused from vscode-maven
+ * https://github.com/microsoft/vscode-maven/blob/2ab8f392f418c8e0fe2903387f2b0013a1c50e78/src/utils/mavenUtils.ts
+ * @param projectFolder
+ */
+async function getLocalGradleWrapper(projectFolder: string): Promise<string | undefined> {
+    const gradlew: string = isWin() ? "gradlew.bat" : "gradlew";
+    // walk up parent folders
+    let current: string = projectFolder;
+    while (Path.basename(current)) {
+        const potentialGradlewFullPath: string = Path.join(current, gradlew);
+        const potentialGradlewPath: string = Path.join(current);
+        if (await pathExists(potentialGradlewFullPath)) {
+            return potentialGradlewPath;
+        }
+        current = Path.dirname(current);
+    }
+    return undefined;
+}
+
+/**
+ * Search for potential Gradle wrapper, return undefined if it does not exist
+ * Modified from vscode-maven, see getLocalMavenWrapper method above
+ * @param projectFolder
+ */
+async function getLocalMavenWrapper(projectFolder: string): Promise<string | undefined> {
```

Comment:
Same comments for `getLocalGradleWrapper` apply to this method as well. These two methods are almost identical.

Reply:
Addressed. Thanks